packages/riverpod_lint/lib/src/lints/avoid_buildContext_in_provider.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/error/listener.dart';
+import 'package:custom_lint_builder/custom_lint_builder.dart';
+import '../riverpod_custom_lint.dart';
+
+
+class AvoidBuildContextInProviders extends RiverpodLintRule {
+  const AvoidBuildContextInProviders() : super(code: _code);
+
+  static const _code = LintCode(
+    name: "avoid_buildContext_in_providers",
+    problemMessage: "Avoid passing BuildContext to providers",
+  );
+
+  @override
+  void run(CustomLintResolver resolver, ErrorReporter reporter,
+      CustomLintContext context) {
+    riverpodRegistry(context).addProviderListenableExpression((expression) {
```

Comment:
Rather than checking ProviderListenableExpression, you probably should be checking Stateful/StatelessProviderDeclarations, and see if their public API has a BuildContext

Reply:
I'm struggling on how to write test for this lint. Actually i have something like this and i don't really know if i'm checking the correct type.. Sorry for disturbing you with that
```Dart
    riverpodRegistry(context).addStatefulProviderDeclaration((declaration) {
      final buildMethod = declaration.buildMethod;
      if(buildMethod.parameters == null) return;
      for (final parameter in buildMethod.parameters!.parameters){
        if(parameter.declaredElement!.typeParameters.first.name == 'BuildContext'){
          reporter.reportErrorForNode(_code, declaration.node);
        }
      }
    });
```

Reply:
See the riverpod_lint_flutter_test folder for reference on how to test things

Generally, you simply write code that should warn (or not warn) in there.

Reply:
Don't check types by their name though. See `TypeChecker`. And `typeParameters` isn't what you want, that's for generics, not function parameters.